simple_salesforce/api.py

```diff
@@ -774,22 +803,12 @@ def __init__(self, username, password, security_token, sandbox=False,
 
 def _exception_handler(result, name=""):
```

Comment:
`def _exception_handler` could be removed totally. I left it here not to brake any other forks using it. It's private, but I assume using a proper deprecation process would be best.